### DISCUSSION TITLE
Getting different responses from web ui vs api for same model and prompt. Is this expected?

### DISCUSSION THREAD

**Discussion by twelchdev** [oobabooga/text-generation-webui#1667] [Q&A] [answered]
I am starting the server using
`call python server.py --auto-devices --api --model gpt4-x-alpaca-13b-native-4bit-128g --no-stream`

Then I use the following code to send a request to the api:
````
HOST = 'localhost:5000'
URI = f'http://{HOST}/api/v1/generate'

def get_response_for_prompt(prompt,max_retries=20, delay=5):
    request = {
        'prompt': prompt,
        'max_new_tokens': 200,
        'do_sample': True,
        'temperature': 0.7,
        'top_p': 0.1,
        'typical_p': 1,
        'repetition_penalty': 1.18,
        'encoder_repetition_penalty': 1,
        'top_k': 40,
        'min_length': 0,
        'no_repeat_ngram_size': 0,
        'num_beams': 1,
        'penalty_alpha': 0,
        'length_penalty': 1,
        'early_stopping': False,
        'seed': -1,
        'add_bos_token': True,
        'truncation_length': 2048,
        'ban_eos_token': False,
        'skip_special_tokens': True,
        'stopping_strings': [],
        'mode': 'chat',
        'chat_generation_attempts': 1,
        'stop_at_newline': False,
        'end_of_turn': ''
    }

    retries = 0
    while retries < max_retries:
        try:
            response = requests.post(URI, json=request)
            if response.status_code == 200:  # Server is accessible
                print("Server is running.")
                result = response.json()['results'][0]['text']
                print(prompt + result)
                return True
        except requests.exceptions.RequestException:
            pass  # Ignore the exception
        print("Request failed trying again ")
        retries += 1
        time.sleep(delay)
````

 ````
prompt = "Hello how are you today? Please list a recipe for making chicken noodle soup\n"
get_response_for_prompt(prompt)
````
   
When I send a request through api I get the following output:
**släktet: Chicken, Noodles, Soup**
        
But when giving the same prompt through web-ui I get:
**Hi! I'm doing well, thank you for asking. Here's a delicious chicken noodle soup recipe: 1. In a large pot, bring water to a boil and add salt. 2. Add the pasta and cook according to package instructions until al dente. 3. While the pasta cooks, heat butter or oil in a separate pan over medium heat. 4. Sauté onions, leeks, and celery in the pan until softened. 5. Add minced garlic and cook for about one minute. 6. Stir in vegetable broth, chicken broth, and chopped chicken into the pan. 7. Simmer the mixture while waiting for the pasta to finish cooking. 8. Drain the cooked pasta and add it to the soup base. Mix well. 9. Serve immediately, garnished with pars**

I tried to make sure my parameters match online exactly, but it did not resolve the issue.  Couldn't find much documentation on the api besides the api_example.py so coming here in desperation.

Is the api supported for chat mode? Are there links to examples/documentation I am missing?
<img width="520" alt="Screenshot 2023-04-29 181333" src="https://user-images.githubusercontent.com/115600237/235330959-42bc3363-5180-4f32-b649-1d32f468247b.png">

**Comment by hypersniper05** [comment]
maybe the key prompts are missing?

**Reply by twelchdev** [reply, marked as answer]
Thanks for replying.  I discovered I am able to get matching prompts if I switch to using `'mode' : 'instruct'` and give an assistant character then adding instruction like
_### Instruction: 
Respond to the following as the Assistant:
Hello how are you today? Please list a recipe for making chicken noodle soup_

And only printing the result in the function where I make the api call.  Guessing chat isn't fully supported as there were[ some issues raised about using chat before](https://github.com/oobabooga/text-generation-webui/discussions/252)  was curious if it was resolved tho as thread is older.
